pkg/stub/handler.go

```diff
@@ -61,6 +65,14 @@ func (h *Handler) deleteDNS(dns *dnsv1alpha1.ClusterDNS) error {
 }
 
 func (h *Handler) syncDNSUpdate(dns *dnsv1alpha1.ClusterDNS) error {
+	if dns.Spec.ClusterIP == nil {
+		// Check for default cluster ip.
+		ipaddr, err := util.ClusterDNSIP(k8sclient.GetKubeClient())
+		if err == nil {
+			logrus.Infof("Using default cluster dns ip address %s", ipaddr)
+			dns.Spec.ClusterIP = &ipaddr
+		}
```

Comment:
Forgot to log when `err != nil`

Reply:
Should probably just throw the error out and let the handler re-queue it

Reply:
We were generating too many messages before the resync period changed but I can add that in now.

Reply:
Oops, I should have reviewed this PR earlier. 
This will stamp ClusterIP from CVO (if not set) for all dns custom resources. This seems like a hack and we are not leveraging any of the CRD/CR stuff for cluster dns and will make it hard to support multiple ClusterDNS configurations in the future.
Why not do this instead:
- CVO has the needed info, populate the ClusterDNS custom resource with the necessary info from CVO and then create the custom resource (probably we need to do this in CVO repo as it owns the responsibility of running the operator config). Code [1] and [2] ensures everything works as expected. 
[1]  https://github.com/openshift/cluster-dns-operator/blob/master/pkg/manifests/manifests.go#L136
[2]  https://github.com/openshift/cluster-dns-operator/blob/master/pkg/manifests/manifests.go#L81

@ironcladlou @ramr ^^